Legobot/Storage/Local.py

```diff
@@ -0,0 +1,55 @@
+import os
+import sys
+import logging
+from Legobot import StorageInterface
+
+logger = logging.getLogger(__name__)
+
+class Key:
+    """
+    Creates a key object that is consumed by storage interface
+    """
+
+    def __init__(self, path):
+        self.path = path
+
+    def exists(self):
+        return os.path.isfile(self.path)
+
+class Local:
+    """
+    Provides a local key/value storage interface
+    """
+
+    def __init__(self, datadir):
+        self.datadir = os.path.abspath(datadir)
+        logger.info("Initialized local storage at {0}".format(self.datadir))
+
+    def set(self, keypath, value, log_transation=True):
+        logger.info("Writing value {0} to key {1}".format(value, keypath))
+        # prevent arbitrary writes to anywhere on filesystem
+        # I think this is still vulnerable to dir walks like ../../../etc/shadow
```

Comment:
Pre-process the string to remove all instances of "../" to avoid this vulnerability.

Reply:
Although, even then, it's probably vulnerable to symlinks, so really, you should ship this with a disclaimer about setting the file permissions so the bot only has access to its own directory as a non-priviliged user.

Reply:
I found this with a quick Google: https://stackoverflow.com/questions/21411904/python-how-to-check-if-path-is-a-subpath

Reply:
Also see os.path.realpath and os.path.normpath to get rid of sym-links and "../" stuff as a pre-processing step. In summary, you would want to call os.path.realpath and os.path.normpath on the passed-in path, and then check that it's a subpath.

Reply:
storage should have never been committed. it wasn't ready yet. I'll `git rm` it before the release.